xray/plot/plot.py

```diff
@@ -55,6 +55,17 @@ def _load_default_cmap(fname='default_colormap.csv'):
     return LinearSegmentedColormap.from_list('viridis', cm_data)
 
 
+def _title_for_slice(darray):
+    '''
+    If the dataarray comes from a slice we can show that info in the title
+    '''
+    title = []
+    for key, value in darray.coords.items():
+        if value.size == 1:
```

Comment:
Yeah, why don't you grab the titles, then squeeze the DataArray? If it can't be squeezed to a 2d array, then raise that error.